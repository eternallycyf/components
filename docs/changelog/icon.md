---
title: Icon
nav:
  title: changelog
  order: 999
tocDepth: 2
---

## Icon

<embed src="../../packages/icon/CHANGELOG.md"></embed>
