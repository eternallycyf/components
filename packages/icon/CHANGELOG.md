# Changelog

## Version&nbsp;1.0.0

<sup>Released on **2024-04-27**</sup>

#### ✨ 新特性

- Add Icon component.

<br/>

<details>
<summary><kbd>Improvements and Fixes</kbd></summary>

#### What's improved

- Add Icon component ([33e4af2](https://github.com/eternallycyf/components/commit/33e4af2))

</details>
