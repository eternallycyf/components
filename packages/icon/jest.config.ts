import { Config } from '@umijs/test';
import base from '../../jest.config.base';

const packageName = '@ims-view/icon';

const root = '<rootDir>/packages/icon';

const config: Config.InitialOptions = {
  ...base,
  rootDir: '../..',
  roots: [root],
  displayName: packageName,
};

export default config;
