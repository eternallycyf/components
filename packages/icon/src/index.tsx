export { default as Icon } from './Icon';
export { default as createFrontIconfont } from './Icon/createFrontIconfont';
export * from './Icon/icons';
export * from './Icon/interface';
