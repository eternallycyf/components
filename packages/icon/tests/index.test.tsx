import { render } from '@testing-library/react';

import { Icon } from '@ims-view/icon';
import React from 'react';

test('Icon', () => {
  const { container } = render(<Icon />);

  expect(container).toMatchSnapshot();
});
