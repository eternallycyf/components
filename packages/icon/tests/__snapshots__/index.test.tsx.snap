// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon 1`] = `
<div>
  <svg
    class="icon"
    fill="currentColor"
    height="1em"
    width="1em"
  />
</div>
`;
